xds/internal/testutils/resource_watcher.go

```diff
@@ -37,7 +37,8 @@ type TestResourceWatcher struct {
 
 // OnUpdate is invoked by the xDS client to report the latest update on the resource
 // being watched.
-func (w *TestResourceWatcher) OnUpdate(data xdsresource.ResourceData) {
+func (w *TestResourceWatcher) OnUpdate(data xdsresource.ResourceData, onDone xdsresource.DoneNotifier) {
+	defer onDone.OnDone()
```

Comment:
This is a test so probably doesn't matter, but doesn't this need to wait for the component reading from UpdateCh/ErrorCh/ResourceDoesNotCh? Probably not because probably just used for test assertions.

Reply:
Yeah, most tests simply verify that one of those events happen and if it is an update, then they probably verify that the update matches expectations.

And its not going to be straightforward to ensure that the appropriate channel is read before invoking the `onDone` callback. Could be done, by also sending the callback in the appropriate channel and having the test invoke it once they have read from the channel. But I feel that's too much work, and will mostly be unrelated to the test under question.

Reply:
Sounds good.